rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -113,14 +113,18 @@ documentation, definitions, references, implementations and compiler errors and
 }
 data LanguageService
     = parser(Parser parser)
-    | summarizer(Summarizer summarizer
+    | analyzer(Summarizer summarizer // triggered on change, or if information is requested
+        , bool providesDocumentation = true
+        , bool providesDefinitions = true
+        , bool providesReferences = true
+        , bool providesImplementations = true)
+    | builder(Summarizer builder // triggered only on save
```

Comment:
if we decide to drop the builder is run automatically, we can also drop the kw params to this one.

Reply:
Dropped the params of builder

Reply:
added better documentation, and dropped the last params of the deprecated old summarizer function, since they could not be mapped onto the builder anymore. This will only produce some warnings in client code, not errors.